osu.Framework/Screens/ScreenStack.cs

```
@@ -101,6 +101,12 @@ internal void Push(IScreen source, IScreen newScreen)
 
             var newScreenDrawable = newScreen.AsDrawable();
 
+            if (newScreenDrawable.IsLoaded)
+                throw new InvalidOperationException("A screen should not be loaded before being pushed.");
```

We definitely don't want this. Make this call OnEntering() instead?

---

If this is actually a possible workflow, it needs a test written. As discussed, it's likely not, since screens are disposed on exiting.

---

If you're going to be defining this behaviour _now_, then the `Push()` xmldoc needs to state this.

Also there needs to be a test for this.

---

Previously it would silently fail. Now it more correctly throws. Documentation and test added.